extensions/common/request_info.cc

```diff
@@ -0,0 +1,459 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License") {}
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/common/request_info.h"
+
+#include "extensions/common/util.h"
+
+// WASM_PROLOG
+#ifndef NULL_PLUGIN
+#include "proxy_wasm_intrinsics.h"
+
+#else  // NULL_PLUGIN
+
+#include "extensions/common/wasm/null/null_plugin.h"
+
+using Envoy::Extensions::Common::Wasm::HeaderMapType;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::getHeaderMapValue;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::getStringValue;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::getValue;
+
+#endif  // NULL_PLUGIN
+
+namespace Wasm {
+namespace Common {
+
+const char RbacFilterName[] = "envoy.filters.http.rbac";
+const char RbacPermissivePolicyIDField[] = "shadow_effective_policy_id";
+const char RbacPermissiveEngineResultField[] = "shadow_engine_result";
+const char kBlackHoleCluster[] = "BlackHoleCluster";
+const char kPassThroughCluster[] = "PassthroughCluster";
+const char kInboundPassthroughClusterIpv4[] = "InboundPassthroughClusterIpv4";
+const char kInboundPassthroughClusterIpv6[] = "InboundPassthroughClusterIpv6";
+const char B3TraceID[] = "x-b3-traceid";
+const char B3SpanID[] = "x-b3-spanid";
+const char B3TraceSampled[] = "x-b3-sampled";
+
+namespace {
+
+// Extract service name from service host.
+void extractServiceName(const std::string& host,
+                        const std::string& destination_namespace,
+                        std::string* service_name) {
+  auto name_pos = host.find_first_of(".:");
+  if (name_pos == std::string::npos) {
+    // host is already a short service name. return it directly.
+    *service_name = host;
+    return;
+  }
+  if (host[name_pos] == ':') {
+    // host is `short_service:port`, return short_service name.
+    *service_name = host.substr(0, name_pos);
+    return;
+  }
+
+  auto namespace_pos = host.find_first_of(".:", name_pos + 1);
```

Comment:
Can you add low level unit tests for request_info.cc?

Which part specifically fix the SD log correlation issues?